### DISCUSSION TITLE
Time-based DynamicPartitionsDefinition

### DISCUSSION THREAD

**Discussion by krgn** [dagster-io/dagster#18889] [Q&A]
Hi all, 

I'm working on a case where I would like to model partitions based on a hierarchy with the top-level partition being a time window (daily, currently), and the lower-level partitions being dynamically loaded for each of the top-level partition time slots. 

For each day, I want the list of sub-partitions to correspond to currently active customers. It might look and evolve something like this:

* 2023-12-24
  + customer B
  + customer C
* 2023-12-23
  + customer A
  + customer B
  + customer C
* 2023-12-22
  + customer A
  + customer B

I am experimenting with a combination of `DailyPartitionsDefinition` and `DynamicPartitionsDefinition` in a  `MultiPartitionsDefinition` like so:

```python
customer_partitions_def = DynamicPartitionsDefinition(name="customers")
multi_partitions_def = MultiPartitionsDefinition(
    {
        "1_date": DailyPartitionsDefinition(start_date="2023-12-20"),
        "2_customer": customer_partitions_def,
    }
)
```

However, in the sensor which loads the customers into the `DynamicPartitionsDefinition`, I do not have access to the current "date" partition, and thus no way of knowing which snapshot of customer list to load there.

Could anybody point me to a possible solution how to hack together a partitioner which models this relationship? I'm looking at building something completely custom, but maybe there is a more obvious solution?

Thanks, and happy holidays! 🙏🏼

**Comment by Sharpieman20** [comment]
Have a very similar usecase currently, wondering if this is more feasable with the updated Partition api since this question was originally asked.
